### DISCUSSION TITLE
feat: lite version (frontend only)

### DISCUSSION THREAD

**Discussion by tmc** [open-webui/open-webui#5993] [General]
I really enjoyed this project most when it could simply be a frontend for ollama. Please make it an option to reduce complexity and not require a backend process.

`npm i; npm run dev` -> working UI was wonderfully easy, and this has been lost.

**Comment by skyler14** [comment]
I didn't get to use it back then, basically are you saying that you just connected the front end to some external API and then used it as a thin-client. Cuz if so that would actually be pretty sweet and I'd second that kind of capability

**Comment by tjbck** [comment]
Not on the short term roadmap but PR welcome!

**Comment by justinh-rahb** [comment]
> Not on the short term roadmap but PR welcome!

To clarify, our goal is to enable a frontend-only build within our existing codebase, without the need to create a separate project and remove all backend code permanently. An ideal solution would be a build-time flag for the frontend that disables backend-dependent functionality.

Additionally, this flag should save all preferences and settings to LocalStorage and allow Ollama/OpenAI API calls to be sent directly, bypassing the need for backend proxying. This will allow for a more streamlined and efficient frontend-only build based off the same codebase.

**Reply by skyler14** [reply]
> > Not on the short term roadmap but PR welcome!
> 
> To clarify, our goal is to enable a frontend-only build within our existing codebase, without the need to create a separate project and remove all backend code permanently. An ideal solution would be a build-time flag for the frontend that disables backend-dependent functionality.
> 
> Additionally, this flag should save all preferences and settings to LocalStorage and allow Ollama/OpenAI API calls to be sent directly, bypassing the need for backend proxying. This will allow for a more streamlined and efficient frontend-only build based off the same codebase.

Has this been added into the repo?

**Reply by yourfavtheo** [reply]
From what I see from the repo it seems that frontend mode only is still not suported... Following up on what @skyler14 said : Are you planning on adding the possibility of using "light" version of openwebui, with only the ui and not the backend.

**Comment by askarbozcan** [comment]
Bumping this.

**Comment by s4m13337** [comment]
Has this been implemented or is it still not supported? I'm really interested in having a thin client open-webui setup that I can connect to my ollama/llama.cpp instance on a different machine.
